control-plane/agents/src/bin/ha/node/server.rs

```diff
@@ -432,10 +432,23 @@ impl ParsedUri {
 
 impl EventWithMeta for NodeAgentSvc {
     fn event(&self, event_action: EventAction, meta: EventMeta) -> EventMessage {
+        let volume_id = meta
```

Comment:
Seems rather complicated, could we change the event fn to receive also optional nqn directly?

Reply:
We are using the Event trait here. We might have to add another event trait to pass the extra argument nqn.

Reply:
A custom function here would probably suffice?

Reply:
Added another private trait for this event.